src/NServiceBus.Core/Persistence/PersistenceDefinition.cs

```diff
@@ -68,6 +68,14 @@ public bool HasSupportFor(Storage storage)
             return storageToActionMap.ContainsKey(typeof(T));
         }
 
+        /// <summary>
+        /// True if supplied storage is supported
+        /// </summary>
+        public bool HasSupportFor(Type storageType) 
```

Comment:
I think if it is fine for a given persistence assembly to support a subset of features, it should be also possible to query what that subset is. Although it would not be an API that users use, it is a part of a public contract of a persistence lib.